aiidalab_widgets_base/structures.py

```diff
@@ -8,17 +8,22 @@
 import ipywidgets as ipw
 from traitlets import Instance, Int, List, Unicode, Union, dlink, link, default, observe
 
-from sklearn.decomposition import PCA
+from optimade_client.query_filter import OptimadeQueryFilterWidget
+from optimade_client.query_provider import OptimadeQueryProviderWidget
```

Comment:
I could, in theory, move these imports down into the widget itself. They would then be imported _only_ together with the widget, but they would also be imported _every time_ with the widget ...

Reply:
It is fine like this I think. We typically move all the imports to the top.